esp-hal/build.rs

```diff
@@ -122,7 +122,43 @@ fn main() -> Result<(), Box<dyn Error>> {
             Some(Validator::Enumeration(
                 vec![String::from("quad"), String::from("octal")]
             )),
-        )
+        ),
+        (
+            "generate-esp-idf-app-desc",
+            "Generate esp-idf compatible app descriptor.",
+            Value::Bool(false),
+            None
+        ),
+        (
+            "esp-idf-build-time",
```

Comment:
I'd strongly suggest not calling these esp-idf-*, even if that's how they show up in the bootloader's output. Also, can we at least default to the current date and time?

Reply:
My idea was to let the esp-generate's generated build.rs populate these fields

Reply:
Do you have an example of how such a `build.rs` might look like? It would be great if we could try this PR without trying to come up with a solution you already have.

Reply:
yeah - that's a tricky thing - better to check the feasibility before 👍

Reply:
https://github.com/bjoernQ/app-desc-test

I'm not sure if having an xtask generated is a good thing or not. We could do more useful things with it and users might need/want it sooner or later - but it's also a bit more to learn for beginners. If the build timestamp is the only use for this it's kind of overkill

Unfortunately, I don't see good alternatives

Using current date/time as the default for the config options makes it a random point in time between creation of the project and now and makes things "look correct" while it's not

I could think of linker-magic but that will leave its scary traces in the generated build.rs